File: Codigos/Ejercicio3.java
import java.util.ArrayList;
import java.util.Scanner;
 
public class Ejercicio3 {
    public static void main(String[] args) {
        ArrayList<String> productos = new ArrayList<>();
        Scanner sc = new Scanner(System.in);
 
        System.out.print("¿Cuantos productos desea registrar? ");
        int n = sc.nextInt();
 
        // Registrar productos
        for (int i = 0; i < n; i++) {
            System.out.print("Producto " + (i+1) + ": ");
            productos.add(sc.next());
        }
 
        // Buscar producto
        System.out.print("Registro y Búsqueda de Productos(Java) " );
        System.out.print("\nIngrese el producto a buscar: ");
        String buscar = sc.next();
 
        int idx = productos.indexOf(buscar);
        if (idx != -1) {
            System.out.println("Producto ENCONTRADO en posicion: " + idx);
        } else {
            System.out.println("Producto NO encontrado.");
        }
        sc.close();
    }
}

File: Codigos/Ejercicio3.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;
 
int main() {
    vector<string> productos;
    string producto, buscar;
    int n;
 
    cout << "¿Cuantos productos desea registrar? ";
    cin >> n;
 
    // Registrar productos
    for (int i = 0; i < n; i++) {
        cout << "Producto " << i+1 << ": ";
        cin >> producto;
        productos.push_back(producto);
    }
 
    // Buscar producto
    cout << "\nIngrese el producto a buscar: ";
    cin >> buscar;
 
    auto it = find(productos.begin(), productos.end(), buscar);
    if (it != productos.end()) {
        cout << "Producto ENCONTRADO en la posicion: "
             << distance(productos.begin(), it) << endl;
    } else {
        cout << "Producto NO encontrado." << endl;
    }
    return 0;
}
